### PR TITLE
[Doc] Update data partitioning docs

### PR DESCRIPTION
**Pull request by bobhan1**
# Versions 

- [x] dev
- [x] 3.0
- [x] 2.1
- [x] 2.0

# Languages

- [x] Chinese
- [x] English

https://github.com/apache/doris/pull/43373

### CODE REVIEW

File: i18n/zh-CN/docusaurus-plugin-content-docs/current/table-design/data-partitioning/dynamic-partitioning.md [hunk 1]
```diff
@@ -0,0 +1,454 @@
...
+-->
+
+
+开启动态分区的表，将会按照设定的规则添加、删除分区，从而对表的分区实现生命周期管理(TTL)，减少用户的使用负担。
+
+动态分区只支持在 DATE/DATETIME 列上进行 Range 类型的分区。
+
+动态分区适用于分区列的时间数据随现实世界同步增长的情况。此时可以灵活的按照与现实世界同步的时间维度对数据进行分区。
+
+对于更为灵活，适用场景更多的数据入库分区，请参阅[自动分区](#自动分区)功能。
```
Comment (yagagagaga) [inline i18n/zh-CN/docusaurus-plugin-content-docs/current/table-design/data-partitioning/dynamic-partitioning.md:34]:
```suggestion
对于更为灵活，适用场景更多的数据入库分区，请参阅[自动分区](./auto-partitioning.md)功能。
```

File: i18n/zh-CN/docusaurus-plugin-content-docs/current/table-design/data-partitioning/manual-partitioning.md [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+
+## Range 分区
+
+分区列通常为时间列，以方便的管理新旧数据。Range 分区支持的列类型 DATE,  DATETIME, TINYINT, SMALLINT, INT, BIGINT, LARGEINT。
+
+**分区信息，支持四种写法：**            
+
+1.  FIXED RANGE：定义分区的左闭右开区间。  
+
+```Plain
```
Comment (yagagagaga) [inline i18n/zh-CN/docusaurus-plugin-content-docs/current/table-design/data-partitioning/manual-partitioning.md:45]:
```suggestion
```sql
```

File: i18n/zh-CN/docusaurus-plugin-content-docs/current/table-design/data-partitioning/manual-partitioning.md [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+    PARTITION `p201702` VALUES LESS THAN ("2017-03-01"),
+    PARTITION `p201703` VALUES LESS THAN ("2017-04-01"),
+    PARTITION `p2018` VALUES [("2018-01-01"), ("2019-01-01")),
+    PARTITION `other` VALUES LESS THAN (MAXVALUE)
+)
+```
+
+3. BATCH RANGE：批量创建数字类型和时间类型的 RANGE 分区，定义分区的左闭右开区间，设定步长。 
+
+```Plain
```
Comment (yagagagaga) [inline i18n/zh-CN/docusaurus-plugin-content-docs/current/table-design/data-partitioning/manual-partitioning.md:89]:
```suggestion
```sql
```

File: i18n/zh-CN/docusaurus-plugin-content-docs/current/table-design/data-partitioning/manual-partitioning.md [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+(
+    PARTITION `p201701` VALUES [("2017-01-01"),  ("2017-02-01")),
+    PARTITION `p201702` VALUES [("2017-02-01"), ("2017-03-01")),
+    PARTITION `p201703` VALUES [("2017-03-01"), ("2017-04-01"))
+)
+```
+
+2. LESS THAN：仅定义分区上界。下界由上一个分区的上界决定。 
+
+```Plain
```
Comment (yagagagaga) [inline i18n/zh-CN/docusaurus-plugin-content-docs/current/table-design/data-partitioning/manual-partitioning.md:66]:
```suggestion
```sql
```

File: i18n/zh-CN/docusaurus-plugin-content-docs/current/table-design/data-partitioning/manual-partitioning.md [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+(
+    PARTITION `p_cn` VALUES IN ("Beijing", "Shanghai", "Hong Kong"),
+    PARTITION `p_usa` VALUES IN ("New York", "San Francisco"),
+    PARTITION `p_jp` VALUES IN ("Tokyo")
+)
+```
+
+List 分区也支持多列分区，示例如下：
+
+```Plain
```
Comment (yagagagaga) [inline i18n/zh-CN/docusaurus-plugin-content-docs/current/table-design/data-partitioning/manual-partitioning.md:147]:
```suggestion
```sql
```

File: versioned_sidebars/version-2.0-sidebars.json [hunk 6]
```diff
@@ -66,7 +66,19 @@
...
                 },
                 "table-design/row-store",
-                "table-design/data-partition",
+                {
+                    "type": "category",
+                    "label": "Data Partitioning",
+                    "items": [
+                        "table-design/data-partitioning/basic-concepts",
+                        "table-design/data-partitioning/manual-partitioning",
+                        "table-design/data-partitioning/dynamic-partitioning",
```
Comment (yagagagaga) [inline versioned_sidebars/version-2.0-sidebars.json:75]: The file name does not correspond to the directory

**Review by yagagagaga — CHANGES_REQUESTED** [review]

File: versioned_sidebars/version-2.0-sidebars.json [hunk 7]
```diff
@@ -66,7 +66,19 @@
...
+                        "table-design/data-partitioning/auto-partitioning",
```
Comment (yagagagaga) [inline versioned_sidebars/version-2.0-sidebars.json:76]: The file name does not correspond to the directory

**Review by yagagagaga — CHANGES_REQUESTED** [review]

File: i18n/zh-CN/docusaurus-plugin-content-docs/version-2.0/table-design/data-partitioning/common-issues.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+当遇到这个错误是，通常是 BE 在创建数据分片时遇到了问题。可以参照以下步骤排查：
+
+-   在 fe.log 中，查找对应时间点的 `Failed to create partition` 日志。在该日志中，会出现一系列类似 `{10001-10010}` 字样的数字对。数字对的第一个数字表示 Backend ID，第二个数字表示 Tablet ID。如上这个数字对，表示 ID 为 10001 的 Backend 上，创建 ID 为 10010 的 Tablet 失败了。
+
+-   前往对应 Backend 的 be.INFO 日志，查找对应时间段内，tablet id 相关的日志，可以找到错误信息。
+
+-   以下罗列一些常见的 tablet 创建失败错误，包括但不限于：
+
+    -   BE 没有收到相关 task，此时无法在 be.INFO 中找到 tablet id 相关日志或者 BE 创建成功，但汇报失败。以上问题，请参阅 [安装与部署](../install/cluster-deployment/standard-deployment) 检查 FE 和 BE 的连通性。
```
Comment (yagagagaga) [inline i18n/zh-CN/docusaurus-plugin-content-docs/version-2.0/table-design/data-partitioning/common-issues.md:48]:
```suggestion
    -   BE 没有收到相关 task，此时无法在 be.INFO 中找到 tablet id 相关日志或者 BE 创建成功，但汇报失败。以上问题，请参阅 [安装与部署](../../install/cluster-deployment/standard-deployment) 检查 FE 和 BE 的连通性。
```

File: i18n/zh-CN/docusaurus-plugin-content-docs/version-2.0/table-design/data-partitioning/common-issues.md [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+    -   预分配内存失败。可能是表中一行的字节长度超过了 100KB。
+
+    -   `Too many open files`。打开的文件句柄数超过了 Linux 系统限制。需修改 Linux 系统的句柄数限制。
+
+如果创建数据分片时超时，也可以通过在 fe.conf 中设置 `tablet_create_timeout_second=xxx` 以及 `max_create_table_timeout_second=xxx` 来延长超时时间。其中 `tablet_create_timeout_second` 默认是 1 秒，`max_create_table_timeout_second` 默认是 60 秒，总体的超时时间为 min(tablet_create_timeout_second * replication_num, max_create_table_timeout_second)，具体参数设置可参阅 [FE 配置项](../admin-manual/config/fe-config) 。
```
Comment (yagagagaga) [inline i18n/zh-CN/docusaurus-plugin-content-docs/version-2.0/table-design/data-partitioning/common-issues.md:54]:
```suggestion
如果创建数据分片时超时，也可以通过在 fe.conf 中设置 `tablet_create_timeout_second=xxx` 以及 `max_create_table_timeout_second=xxx` 来延长超时时间。其中 `tablet_create_timeout_second` 默认是 1 秒，`max_create_table_timeout_second` 默认是 60 秒，总体的超时时间为 min(tablet_create_timeout_second * replication_num, max_create_table_timeout_second)，具体参数设置可参阅 [FE 配置项](../../admin-manual/config/fe-config) 。
```

File: i18n/zh-CN/docusaurus-plugin-content-docs/version-2.0/table-design/data-partitioning/common-issues.md [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+
+**3.  建表命令长时间不返回结果。**
+
+-   Doris 的建表命令是同步命令。该命令的超时时间目前设置的比较简单，即（tablet num * replication num）秒。如果创建较多的数据分片，并且其中有分片创建失败，则可能导致等待较长超时后，才会返回错误。
+
+-   正常情况下，建表语句会在几秒或十几秒内返回。如果超过一分钟，建议直接取消掉这个操作，前往 FE 或 BE 的日志查看相关错误。
+
+## 更多帮助
+
+关于数据划分更多的详细说明，我们可以在 [CREATE TABLE](../sql-manual/sql-reference/Data-Definition-Statements/Create/CREATE-TABLE) 命令手册中查阅，也可以在 MySQL 客户端下输入 `HELP CREATE TABLE;` 获取更多的帮助信息。
```
Comment (yagagagaga) [inline i18n/zh-CN/docusaurus-plugin-content-docs/version-2.0/table-design/data-partitioning/common-issues.md:64]:
```suggestion
关于数据划分更多的详细说明，我们可以在 [CREATE TABLE](../../sql-manual/sql-reference/Data-Definition-Statements/Create/CREATE-TABLE) 命令手册中查阅，也可以在 MySQL 客户端下输入 `HELP CREATE TABLE;` 获取更多的帮助信息。
```

File: i18n/zh-CN/docusaurus-plugin-content-docs/current/table-design/data-partitioning/auto-partitioning.md [hunk 11]
```diff
@@ -0,0 +1,280 @@
...
+)
+DISTRIBUTED BY HASH(`TRADE_DATE`) BUCKETS 10
+PROPERTIES (
+  "replication_num" = "1"
+);
+```
+
+
+
+该表内存储了大量业务历史数据，依据交易发生的日期进行分区。可以看到在建表时，我们需要预先手动创建分区。如果分区列的数据范围发生变化，例如上表中增加了 2022 年的数据，则我们需要通过[ALTER-TABLE-PARTITION](../sql-manual/sql-statements/Data-Definition-Statements/Alter/ALTER-TABLE-PARTITION)对表的分区进行更改。如果这种分区需要变更，或者进行更细粒度的细分，修改起来非常繁琐。此时我们就可以使用 AUTO PARTITION 改写该表 DDL。
```
Comment (yagagagaga) [inline i18n/zh-CN/docusaurus-plugin-content-docs/current/table-design/data-partitioning/auto-partitioning.md:77]:
```suggestion
该表内存储了大量业务历史数据，依据交易发生的日期进行分区。可以看到在建表时，我们需要预先手动创建分区。如果分区列的数据范围发生变化，例如上表中增加了 2022 年的数据，则我们需要通过[ALTER-TABLE-PARTITION](../../sql-manual/sql-statements/Data-Definition-Statements/Alter/ALTER-TABLE-PARTITION)对表的分区进行更改。如果这种分区需要变更，或者进行更细粒度的细分，修改起来非常繁琐。此时我们就可以使用 AUTO PARTITION 改写该表 DDL。
```

File: i18n/zh-CN/docusaurus-plugin-content-docs/current/table-design/data-partitioning/auto-partitioning.md [hunk 12]
```diff
@@ -0,0 +1,280 @@
...
+
+## 语法
+
+建表时，使用以下语法填充[CREATE-TABLE](../sql-manual/sql-statements/Data-Definition-Statements/Create/CREATE-TABLE)时的`partition_info`部分：
```
Comment (yagagagaga) [inline i18n/zh-CN/docusaurus-plugin-content-docs/current/table-design/data-partitioning/auto-partitioning.md:81]:
```suggestion
建表时，使用以下语法填充[CREATE-TABLE](../../sql-manual/sql-statements/Data-Definition-Statements/Create/CREATE-TABLE)时的`partition_info`部分：
```

File: i18n/zh-CN/docusaurus-plugin-content-docs/current/table-design/data-partitioning/auto-partitioning.md [hunk 13]
```diff
@@ -0,0 +1,280 @@
...
+为使分区逻辑清晰，Doris 禁止自动分区（Auto Partition）和动态分区（Dynamic Partition）同时作用于一张表上，这种用法容易引发误用，应当以单独的自动分区功能代替。
+
+注意：在 Doris 2.1 的某些早期版本中，该功能未被禁止，但不推荐使用。
+
+## 注意事项
+
+- 如同普通分区表一样，AUTO LIST PARTITION 支持多列分区，语法并无区别。
+- 在数据的插入或导入过程中如果创建了分区，而整个导入过程没有完成（失败或被取消），被创建的分区不会被自动删除。
+- 使用 AUTO PARTITION 的表，只是分区创建方式上由手动转为了自动。表及其所创建分区的原本使用方法都与非 AUTO PARTITION 的表或分区相同。
+- 为防止意外创建过多分区，我们通过[FE 配置项](../admin-manual/config/fe-config)中的`max_auto_partition_num`控制了一个 AUTO PARTITION 表最大容纳分区数。如有需要可以调整该值
```
Comment (yagagagaga) [inline i18n/zh-CN/docusaurus-plugin-content-docs/current/table-design/data-partitioning/auto-partitioning.md:277]:
```suggestion
- 为防止意外创建过多分区，我们通过[FE 配置项](../../admin-manual/config/fe-config)中的`max_auto_partition_num`控制了一个 AUTO PARTITION 表最大容纳分区数。如有需要可以调整该值
```

File: i18n/zh-CN/docusaurus-plugin-content-docs/current/table-design/data-partitioning/auto-partitioning.md [hunk 14]
```diff
@@ -0,0 +1,280 @@
...
+- 向开启了 AUTO PARTITION 的表导入数据时，Coordinator 发送数据的轮询间隔与普通表有所不同。具体请见[BE 配置项](../admin-manual/config/be-config)中的`olap_table_sink_send_interval_auto_partition_factor`。开启前移（`enable_memtable_on_sink_node = true`）后该变量不产生影响。
```
Comment (yagagagaga) [inline i18n/zh-CN/docusaurus-plugin-content-docs/current/table-design/data-partitioning/auto-partitioning.md:278]:
```suggestion
- 向开启了 AUTO PARTITION 的表导入数据时，Coordinator 发送数据的轮询间隔与普通表有所不同。具体请见[BE 配置项](../../admin-manual/config/be-config)中的`olap_table_sink_send_interval_auto_partition_factor`。开启前移（`enable_memtable_on_sink_node = true`）后该变量不产生影响。
```

File: i18n/zh-CN/docusaurus-plugin-content-docs/current/table-design/data-partitioning/auto-partitioning.md [hunk 15]
```diff
@@ -0,0 +1,280 @@
...
+- 在使用[insert-overwrite](../sql-manual/sql-statements/Data-Manipulation-Statements/Manipulation/INSERT-OVERWRITE)插入数据时，如果指定了覆写的 partition，则 AUTO PARTITION 表在此过程中表现得如同普通表，不创建新的分区。
```
Comment (yagagagaga) [inline i18n/zh-CN/docusaurus-plugin-content-docs/current/table-design/data-partitioning/auto-partitioning.md:279]:
```suggestion
- 在使用[insert-overwrite](../../sql-manual/sql-statements/Data-Manipulation-Statements/Manipulation/INSERT-OVERWRITE)插入数据时，如果指定了覆写的 partition，则 AUTO PARTITION 表在此过程中表现得如同普通表，不创建新的分区。
```

File: i18n/zh-CN/docusaurus-plugin-content-docs/version-2.0/table-design/data-partitioning/manual-partitioning.md [hunk 16]
```diff
@@ -0,0 +1,118 @@
...
+
+示例如下：
+
+```sql
+PARTITION BY RANGE(`date`)
+(
+    PARTITION `p201701` VALUES [("2017-01-01"),  ("2017-02-01")),
+    PARTITION `p201702` VALUES [("2017-02-01"), ("2017-03-01")),
+    PARTITION `p201703` VALUES [("2017-03-01"), ("2017-04-01"))
+)
```
[the remark this answers is not in the document]
Comment (bobhan1) [reply i18n/zh-CN/docusaurus-plugin-content-docs/version-2.0/table-design/data-partitioning/manual-partitioning.md:66]: 这个pr先改结构和链接吧，后面再单独提pr改这里面每一个的内容

File: i18n/zh-CN/docusaurus-plugin-content-docs/version-2.1/table-design/data-partitioning/basic-concepts.md [hunk 17]
```diff
@@ -0,0 +1,187 @@
...
+
+- 均匀数据分布：分桶可以将数据均匀地分布在各个桶中，减少了数据集中或倾斜的风险，从而避免了部分节点或存储设备的资源过载问题。
+
+- 减少热点：通过均匀分布数据，分桶有助于减少某些特定节点或分区过度负载的现象，避免热点问题，提升系统的稳定性和处理能力。
+
+- 提高并发性能：分桶帮助提高并发查询的性能，特别是当多个查询请求需要访问相同分区中的不同数据时，分桶的粒度使得系统能够有效地并行处理多个请求，从而提升吞吐量。
+
+## 建表举例
+
+Doris 的建表是一个同步命令，SQL 执行完成即返回结果，命令返回成功即表示建表成功。具体建表语法可以参考[CREATE TABLE](../sql-manual/sql-statements/Data-Definition-Statements/Create/CREATE-TABLE)，也可以通过 `HELP CREATE TABLE` 查看更多帮助。
```
Comment (yagagagaga) [inline i18n/zh-CN/docusaurus-plugin-content-docs/version-2.1/table-design/data-partitioning/basic-concepts.md:77]:
```suggestion
Doris 的建表是一个同步命令，SQL 执行完成即返回结果，命令返回成功即表示建表成功。具体建表语法可以参考[CREATE TABLE](../../sql-manual/sql-statements/Data-Definition-Statements/Create/CREATE-TABLE)，也可以通过 `HELP CREATE TABLE` 查看更多帮助。
```

File: i18n/zh-CN/docusaurus-plugin-content-docs/version-2.1/table-design/data-partitioning/basic-concepts.md [hunk 18]
```diff
@@ -0,0 +1,187 @@
...
+DISTRIBUTED BY HASH(`user_id`) BUCKETS 16
+PROPERTIES
+(
+    "replication_num" = "1"
+);
+```
+
+这里以 AGGREGATE KEY 数据模型为例进行说明。AGGREGATE KEY 数据模型中，所有没有指定聚合方式（SUM、REPLACE、MAX、MIN）的列视为 Key 列。而其余则为 Value 列。
+
+在建表语句的最后 PROPERTIES 中，关于 PROPERTIES 中可以设置的相关参数，可以查看[CREATE TABLE](../sql-manual/sql-statements/Data-Definition-Statements/Create/CREATE-TABLE)中的详细介绍。
```
Comment (yagagagaga) [inline i18n/zh-CN/docusaurus-plugin-content-docs/version-2.1/table-design/data-partitioning/basic-concepts.md:114]:
```suggestion
在建表语句的最后 PROPERTIES 中，关于 PROPERTIES 中可以设置的相关参数，可以查看[CREATE TABLE](../../sql-manual/sql-statements/Data-Definition-Statements/Create/CREATE-TABLE)中的详细介绍。
```

File: i18n/zh-CN/docusaurus-plugin-content-docs/current/table-design/data-partitioning/basic-concepts.md [hunk 19]
```diff
@@ -0,0 +1,187 @@
...
+
+- 均匀数据分布：分桶可以将数据均匀地分布在各个桶中，减少了数据集中或倾斜的风险，从而避免了部分节点或存储设备的资源过载问题。
+
+- 减少热点：通过均匀分布数据，分桶有助于减少某些特定节点或分区过度负载的现象，避免热点问题，提升系统的稳定性和处理能力。
+
+- 提高并发性能：分桶帮助提高并发查询的性能，特别是当多个查询请求需要访问相同分区中的不同数据时，分桶的粒度使得系统能够有效地并行处理多个请求，从而提升吞吐量。
+
+## 建表举例
+
+Doris 的建表是一个同步命令，SQL 执行完成即返回结果，命令返回成功即表示建表成功。具体建表语法可以参考[CREATE TABLE](../sql-manual/sql-statements/Data-Definition-Statements/Create/CREATE-TABLE)，也可以通过 `HELP CREATE TABLE` 查看更多帮助。
```
Comment (yagagagaga) [inline i18n/zh-CN/docusaurus-plugin-content-docs/current/table-design/data-partitioning/basic-concepts.md:77]:
```suggestion
Doris 的建表是一个同步命令，SQL 执行完成即返回结果，命令返回成功即表示建表成功。具体建表语法可以参考[CREATE TABLE](../../sql-manual/sql-statements/Data-Definition-Statements/Create/CREATE-TABLE)，也可以通过 `HELP CREATE TABLE` 查看更多帮助。
```

File: i18n/zh-CN/docusaurus-plugin-content-docs/current/table-design/data-partitioning/basic-concepts.md [hunk 20]
```diff
@@ -0,0 +1,187 @@
...
+DISTRIBUTED BY HASH(`user_id`) BUCKETS 16
+PROPERTIES
+(
+    "replication_num" = "1"
+);
+```
+
+这里以 AGGREGATE KEY 数据模型为例进行说明。AGGREGATE KEY 数据模型中，所有没有指定聚合方式（SUM、REPLACE、MAX、MIN）的列视为 Key 列。而其余则为 Value 列。
+
+在建表语句的最后 PROPERTIES 中，关于 PROPERTIES 中可以设置的相关参数，可以查看[CREATE TABLE](../sql-manual/sql-statements/Data-Definition-Statements/Create/CREATE-TABLE)中的详细介绍。
```
Comment (yagagagaga) [inline i18n/zh-CN/docusaurus-plugin-content-docs/current/table-design/data-partitioning/basic-concepts.md:114]:
```suggestion
在建表语句的最后 PROPERTIES 中，关于 PROPERTIES 中可以设置的相关参数，可以查看[CREATE TABLE](../../sql-manual/sql-statements/Data-Definition-Statements/Create/CREATE-TABLE)中的详细介绍。
```

**Review by yagagagaga — APPROVED** [review]

**Pull request merged**